crates/oxc_linter/src/rules/eslint/accessor_pairs.rs

```diff
@@ -0,0 +1,2074 @@
+        TSTypeLiteral,
+    },
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{ContentEq, GetSpan, Span};
+use rustc_hash::FxHashMap;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::{AstNode, context::LintContext, rule::Rule};
+
+fn setter_without_getter_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Setter is defined without a getter")
+        .with_help("Define a getter for this property")
+        .with_label(span)
+}
+
+fn getter_without_setter_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Getter is defined without a setter")
+        .with_help("Define a setter for this property")
+        .with_label(span)
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct AccessorPairsConfig {
+    #[serde(default = "default_true")]
+    set_without_get: bool,
+    #[serde(default)]
+    get_without_set: bool,
+    #[serde(default = "default_true")]
+    enforce_for_class_members: bool,
+    #[serde(default, rename = "enforceForTSTypes")]
+    enforce_for_ts_types: bool,
+}
+
+fn default_true() -> bool {
+    true
+}
+
+impl Default for AccessorPairsConfig {
+    fn default() -> Self {
+        Self {
+            set_without_get: true,
+            get_without_set: false,
+            enforce_for_class_members: true,
+            enforce_for_ts_types: false,
+        }
+    }
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct AccessorPairs(Box<AccessorPairsConfig>);
+
+impl std::ops::Deref for AccessorPairs {
+    type Target = AccessorPairsConfig;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Enforces getter/setter pairs in objects and classes.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// It's a common mistake in JavaScript to create an object with just a setter
+    /// for a property but never have a corresponding getter defined for it.
+    /// Without a getter, you cannot read the property, so it ends up not being used.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// var o = {
+    ///     set a(value) {
+    ///         this.val = value;
+    ///     }
+    /// };
+    ///
+    /// class C {
+    ///     set a(value) {
+    ///         this.val = value;
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// var o = {
+    ///     set a(value) {
+    ///         this.val = value;
+    ///     },
+    ///     get a() {
+    ///         return this.val;
+    ///     }
+    /// };
+    ///
+    /// class C {
+    ///     set a(value) {
+    ///         this.val = value;
+    ///     }
+    ///     get a() {
+    ///         return this.val;
+    ///     }
+    /// }
+    /// ```
+    AccessorPairs,
+    eslint,
+    pedantic,
```

Comment:
This will fix the test that's failing:

```suggestion
    pedantic,
    config = AccessorPairsConfig,
```